concurrent_map.go

```diff
@@ -5,75 +5,99 @@ import (
 	"sync"
 )
 
-var SHARD_COUNT = 32
+// ShardCount represent the number of shard the map is devided into.
+const ShardCount = 100
 
-// A "thread" safe map of type string:Anything.
-// To avoid lock bottlenecks this map is dived to several (SHARD_COUNT) map shards.
+// ConcurrentMap is a "thread" safe map of type string:Anything.
+// To avoid lock bottlenecks this map is dived to several (ShardCount) map shards.
 type ConcurrentMap []*ConcurrentMapShared
 
-// A "thread" safe string to anything map.
+// ConcurrentMapShared is a "thread" safe string to anything map.
 type ConcurrentMapShared struct {
+	mapKeys      []string
+	maxSize      int
 	items        map[string]interface{}
 	sync.RWMutex // Read Write mutex, guards access to internal map.
 }
 
-// Creates a new concurrent map.
-func New() ConcurrentMap {
-	m := make(ConcurrentMap, SHARD_COUNT)
-	for i := 0; i < SHARD_COUNT; i++ {
-		m[i] = &ConcurrentMapShared{items: make(map[string]interface{})}
+// New creates a new concurrent map.
+func New(maxSize int) ConcurrentMap {
+	m := make(ConcurrentMap, ShardCount)
+	for i := 0; i < ShardCount; i++ {
+		m[i] = &ConcurrentMapShared{
+			maxSize: maxSize,
```

Comment:
I would divide the maxSize by ShardCount and add here the maxSize of one shard.

Reply:
Done.